golem-ts-agentic/packages/golem-ts-sdk/tests/testTypes.ts

```diff
@@ -32,6 +32,33 @@ export type UnionWithOnlyLiterals = 'foo' | 'bar' | 'baz';
 
 export type UnionWithLiterals = 'a' | 'b' | 'c' | boolean;
 
+// tag and `val` are exact as wit-type result
+export type ResultTypeExactBoth =
+  | { tag: 'ok'; val: number }
+  | { tag: 'err'; val: string };
+
+// Result type Kind is similar looking, but it's value instead of val
+export type ResultTypeNonExact =
+  | { tag: 'ok'; value: number }
+  | { tag: 'err'; value: string };
+
+// Result type Kind is similar looking, but it's okValue and errValue instead of val
+export type ResultTypeNonExact2 =
+  | { tag: 'ok'; okValue: number }
+  | { tag: 'err'; errValue: string };
+
+export type ResultTypeInvalid1 =
+  | { tag: 'ok'; okValOpt?: number }
```

Comment:
Cannot be handled because the value or wit-value cannot represent `{tag: 'ok'}`. 

The wit-value is more or less the following.

```
{
      kind: 'result';
      value: {
        ok?: Value;
        err?: Value;
      };
    }
    ```
    
A `{tag: 'ok'}` can be serialized to `{kind: result, value: {ok: undefined}}` but in this case both `ok` and `err` is still simply undefined at the received side.

Hence we reject if user defined a result type with optional values. They can either rename the tag to consider it as a normal variant, which will handle it easily. Or they can keep the tag name and make the values non optional - in error message